spec/features/team_creation_spec.rb

```diff
@@ -0,0 +1,335 @@
+require 'spec_helper'
+require 'rails_helper'
+
+
+
+# require File.join('./app/controllers/application_controller')
+
+
+describe "Team Creation" do
+ 
+  
+   let(:topics){FactoryGirl.create(:topics)}
+
+   before(:each) do
+     FactoryGirl.create(:assignment)
+     FactoryGirl.create(:due_date)
+     FactoryGirl.create(:participants) 
+     FactoryGirl.create(:participants) 
+     FactoryGirl.create(:participants) 
+     FactoryGirl.create(:topics)
+     FactoryGirl.create(:assignmentnode)
+     FactoryGirl.create(:topics,topic_name:"command pattern")
+     FactoryGirl.create(:deadline_type,name:"submission")
+     FactoryGirl.create(:deadline_type,name:"review")
+     FactoryGirl.create(:deadline_type,name:"resubmission")
+     FactoryGirl.create(:deadline_type,name:"rereview")
+     FactoryGirl.create(:deadline_type,name:"metareview")
+     FactoryGirl.create(:deadline_type,name:"drop_topic")
+     FactoryGirl.create(:deadline_type,name:"signup")	
+     FactoryGirl.create(:deadline_type,name:"team_formation")
+   end
+
+
+context'one student who signup for a topic should send an inviatation to the other student who has no topic ' do
+
+   before(:each) do
+   student=User.find_by_name("student2064")  
+   role=student.role
+ 
+   ApplicationController.any_instance.stub(:current_user).and_return(student)
+   ApplicationController.any_instance.stub(:current_role_name).and_return('Student')
+   ApplicationController.any_instance.stub(:current_role).and_return(role)
+ 
+   visit '/student_task/list'
+   #Assignment anme
+   expect(page).to have_content('final2')
+ 
+   click_link 'final2'
+   expect(page).to have_content('Submit or Review work for final2')
+ 
+   click_link 'Signup sheet'
+   expect(page).to have_content('Signup sheet for final2 assignment')
+ 
+   visit '/sign_up_sheet/sign_up?assignment_id=1&id=1'
```

Comment:
Is this necessary? you have already clicked "signup_sheet", what if the assignment_id is not 1?